README.md

```diff
@@ -1 +1,54 @@
-# operator
\ No newline at end of file
+# Volume Replication Operator
+
+## Overview
+
+Volume Replication Operator is a kubernetes operator that provides common and reusable APIs for disaster recovery.
+It is based on [kube-storage/spec](https://github.com/kube-storage/spec) specification and can be used by any storage
+provider.
+
+## Design
+
+Volume Replication Operator follows controller pattern and provides extended APIs for disaster recovery.
+The extended APIs are provided via Custom Resource Definition (CRD). Some CRDs provided by the operator
+are:
+
++ [VolumeReplicationClass](https://github.com/kube-storage/volume-replication-operator/blob/main/config/crd/bases/replication.storage.openshift.io_volumereplicationclasses.yaml)
++ [VolumeReplication](https://github.com/kube-storage/volume-replication-operator/blob/main/config/crd/bases/replication.storage.openshift.io_volumereplications.yaml)
+
+### VolumeReplicationClass
+
+VolumeReplicationClass is a cluster scoped resource that contains driver related configuration parameters.
+It is used by the operator to talk to the CSI backend.
+
+```yaml
+apiVersion: replication.storage.openshift.io/v1alpha1
+kind: VolumeReplicationClass
+metadata:
+  name: volumereplicationclass-sample
+spec:
+  provisioner: example.provisioner.io
+  parameters:
+    secretName: mySecretName
+    secretNamespace: mySecretNamespace
```

Comment:
these fields need to be updated and also mention that the user can add his own key-value pair to the parameters and this will be passed down to the driver.

Reply:
Ack. I'm also considering mentioning the reserved keys here.